qiskit/pulse/schedule.py

```diff
@@ -610,3 +643,44 @@ def bind_parameters(self, *args: List[Union[float, complex]],
     def __call__(self, *args: List[Union[float, complex]],
                  **kwargs: Dict[str, Union[float, complex]]) -> Schedule:
         return self.bind_parameters(*args, **kwargs)
+
+
+def _insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
+    """
+    Using binary search on start times, return the index into `intervals` where the new interval
+    belongs, or raise an error if the new interval overlaps with any existing ones.
+    Args:
```

Comment:
Spaces needed between these sections in docstring?

Reply:
woops, thank you!!!